diesel_codegen_syntex/src/update.rs

```diff
@@ -84,76 +83,26 @@ fn changeset_impl(
     options: &ChangesetOptions,
     model: &Model,
 ) -> Option<P<ast::Item>> {
-    let ref struct_name = model.ty;
-    let pk = model.primary_key_name();
+    let struct_name = model.name;
     let table_name = options.table_name;
-    let attrs_for_changeset = model.attrs.iter().filter(|a| a.column_name.name != pk.name)
-        .collect::<Vec<_>>();
-    let changeset_ty = cx.ty(span, TyKind::Tup(
-        attrs_for_changeset.iter()
-              .map(|a| changeset_ty(cx, span, &options, a))
-              .collect()
-    ));
-    let changeset_body = cx.expr_tuple(span, attrs_for_changeset.iter()
-        .map(|a| changeset_expr(cx, span, &options, a))
-        .collect());
-    quote_item!(cx,
-        impl<'a: 'update, 'update> ::diesel::query_builder::AsChangeset for
-            &'update $struct_name
-        {
-            type Target = $table_name::table;
-            type Changeset = $changeset_ty;
-
-            fn as_changeset(self) -> Self::Changeset {
-                $changeset_body
-            }
-        }
-    )
-}
+    let treat_none_as_null = &options.treat_none_as_null;
+    let struct_ty = &model.ty;
+    let lifetimes = lifetime_list_tokens(&model.generics.lifetimes, span);
 
-fn changeset_ty(
-    cx: &ExtCtxt,
-    span: Span,
-    options: &ChangesetOptions,
-    attr: &Attr,
-) -> P<ast::Ty> {
-    let column = cx.path(span, vec![options.table_name, attr.column_name]);
-    match (options.treat_none_as_null, ty_param_of_option(&attr.ty)) {
-        (false, Some(ty)) => {
-            let inner_ty = inner_changeset_ty(cx, column, &ty);
-            quote_ty!(cx, Option<$inner_ty>)
-        }
-        _ => inner_changeset_ty(cx, column, &attr.ty),
-    }
-}
-
-fn inner_changeset_ty(
-    cx: &ExtCtxt,
-    column: ast::Path,
-    field_ty: &ast::Ty,
-) -> P<ast::Ty> {
-    quote_ty!(cx,
-        ::diesel::expression::predicates::Eq<
-            $column,
-            ::diesel::expression::bound::Bound<
-                <$column as ::diesel::expression::Expression>::SqlType,
-                &'update $field_ty,
-            >,
-        >
-    )
-}
+    let pk = model.primary_key_name();
+    let fields = model.attrs.iter()
+        .filter(|a| a.column_name.name != pk.name)
+        .map(|a| a.to_stable_macro_tokens(cx))
```

Comment:
This line is way better w/ Macros 1.1 because the quoting library I'm using supports repeations so I can just impl `ToTokens` on `Attr` directly and do `$($fields)*` like I would in a normal macro